### PR TITLE
MINOR: [Rust][Docs] Update rust feature support docs

### PR DESCRIPTION
**Pull request by tustvold**
The feature support matrix for Rust is about a year out of date, lets update it :smile: 

Website is https://arrow.apache.org/docs/status.html

### CODE REVIEW

File: docs/source/status.rst [hunk 1]
```diff
@@ -81,11 +81,11 @@ Data Types
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
 | Struct            | ✓     | ✓     | ✓     | ✓          |  ✓    |  ✓    | ✓     |
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
-| Map               | ✓     | ✓     | ✓     | ✓          |       |       | ✓     |
+| Map               | ✓     | ✓     | ✓     | ✓          |       |  ✓    | ✓     |
```
Comment (tustvold) [inline docs/source/status.rst:84]: https://docs.rs/arrow/latest/arrow/array/struct.MapArray.html

File: docs/source/status.rst [hunk 2]
```diff
@@ -81,11 +81,11 @@ Data Types
...
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
-| Dense Union       | ✓     | ✓     | ✓     |            |       |       | ✓     |
+| Dense Union       | ✓     | ✓     | ✓     |            |       |  ✓    | ✓     |
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
-| Sparse Union      | ✓     | ✓     | ✓     |            |       |       | ✓     |
+| Sparse Union      | ✓     | ✓     | ✓     |            |       |  ✓    | ✓     |
```
Comment (tustvold) [inline docs/source/status.rst:88]: https://docs.rs/arrow/latest/arrow/array/struct.UnionArray.html

File: docs/source/status.rst [hunk 3]
```diff
@@ -94,7 +94,7 @@ Data Types
 +===================+=======+=======+=======+============+=======+=======+=======+
 | Dictionary        | ✓     | ✓ (1) | ✓     | ✓ (1)      | ✓ (1) | ✓ (1) | ✓     |
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
-| Extension         | ✓     | ✓     | ✓     |            |       |       | ✓     |
+| Extension         | ✓     | ✓     | ✓     |            |       | ✓     | ✓     |
```
Comment (tustvold) [inline docs/source/status.rst:97]: Not sure about this one, but we support propagating the metadata which seems to be all that is required??

File: docs/source/status.rst [hunk 4]
```diff
@@ -128,7 +128,7 @@ IPC Format
 +-----------------------------+-------+-------+-------+------------+-------+-------+-------+
 | Sparse tensors              | ✓     |       |       |            |       |       |       |
 +-----------------------------+-------+-------+-------+------------+-------+-------+-------+
-| Buffer compression          | ✓     | ✓ (3) | ✓     |            |       |       | ✓     |
+| Buffer compression          | ✓     | ✓ (3) | ✓     |            |       |  ✓    | ✓     |
```
Comment (tustvold) [inline docs/source/status.rst:131]: This was just added by @liukun4515 and @alamb in https://github.com/apache/arrow-rs/pull/2369

File: docs/source/status.rst [hunk 5]
```diff
@@ -156,11 +156,11 @@ Flight RPC
...
 | Flight RPC Transport                       | C++   | Java  | Go    | JavaScript | C#    | Rust  | Julia |
 +============================================+=======+=======+=======+============+=======+=======+=======+
-| gRPC_ transport (grpc:, grpc+tcp:)         | ✓     | ✓     | ✓     |            | ✓     |       |       |
+| gRPC_ transport (grpc:, grpc+tcp:)         | ✓     | ✓     | ✓     |            | ✓     | ✓     |       |
 +--------------------------------------------+-------+-------+-------+------------+-------+-------+-------+
-| gRPC domain socket transport (grpc+unix:)  | ✓     | ✓     | ✓     |            | ✓     |       |       |
+| gRPC domain socket transport (grpc+unix:)  | ✓     | ✓     | ✓     |            | ✓     | ✓     |       |
 +--------------------------------------------+-------+-------+-------+------------+-------+-------+-------+
-| gRPC + TLS transport (grpc+tls:)           | ✓     | ✓     | ✓     |            | ✓     |       |       |
+| gRPC + TLS transport (grpc+tls:)           | ✓     | ✓     | ✓     |            | ✓     | ✓     |       |
```
Comment (tustvold) [inline docs/source/status.rst:163]: Arrow-flight uses tonic which supports [uds](https://github.com/hyperium/tonic/tree/master/examples/src/uds) and [tls](https://github.com/hyperium/tonic/tree/master/examples/src/tls)

File: docs/source/status.rst [hunk 6]
```diff
@@ -40,13 +40,13 @@ Data Types
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
 | UInt8/16/32/64    | ✓     | ✓     | ✓     | ✓          |  ✓    |  ✓    | ✓     |
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
-| Float16           |       |       | ✓     |            |       |       | ✓     |
+| Float16           |       |       | ✓     |            |       |  ✓    | ✓     |
```
Comment (tustvold) [inline docs/source/status.rst:43]: https://docs.rs/arrow/latest/arrow/array/type.Float16Array.html

File: docs/source/status.rst [hunk 7]
```diff
@@ -40,13 +40,13 @@ Data Types
...
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
 | Float32/64        | ✓     | ✓     | ✓     | ✓          |  ✓    |  ✓    | ✓     |
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
 | Decimal128        | ✓     | ✓     | ✓     |            |  ✓    |  ✓    | ✓     |
 +-------------------+-------+-------+-------+------------+-------+-------+-------+
-| Decimal256        | ✓     | ✓     | ✓     |            |  ✓    |       | ✓     |
+| Decimal256        | ✓     | ✓     | ✓     |            |  ✓    |  ✓    | ✓     |
```
Comment (tustvold) [inline docs/source/status.rst:49]: https://docs.rs/arrow/latest/arrow/array/struct.Decimal256Array.html

File: docs/source/status.rst (hunk 4 above)
Comment (alamb) [reply docs/source/status.rst:131]: 🎉

**Review by alamb — APPROVED** [review]
Thanks @tustvold  -- looks great to me. I noticed this the other day and I couldn't figure out where the docs were defined (I was looking in https://github.com/apache/arrow-site) 

BTW i think there is a special script to do the commit/merge in this repo (rather than the big green button). 

I think I caused issues by not using it in the recent past -- for penance I can go figure out the correct invocation and do so for this PR if you would like

**Comment by tustvold** [comment]
> I couldn't figure out where the docs were defined

I used the "Edit this page" button in the top right to work out where it was 
image

> I can go figure out the correct invocation and do so for this PR if you would like

That would be amazing, thank you

**Comment by alamb** [comment]
```shell
(arrow_dev) alamb@MacBook-Pro-8:~/Software/arrow$ ./dev/merge_arrow_pr.sh 13879
ARROW_HOME = /Users/alamb/Software/arrow/dev
PROJECT_NAME = arrow

=== Pull Request #13879 ===
title	MINOR: [Rust][Docs] Update rust feature support docs
source	tustvold/update-rust-feature-support
target	master
url	https://api.github.com/repos/apache/arrow/pulls/13879
Minor PR.  Please ensure it meets guidelines for minor.


Proceed with merging pull request #13879? (y/n): y
Author 1: Raphael Taylor-Davies <r.taylordavies@googlemail.com>
Pull request #13879 merged!
Merge hash: a270afc946398a0279b1971a315858d8b5f07e2d
Minor PR.  No JIRA issue to update.

(arrow_dev) alamb@MacBook-Pro-8:~/Software/arrow$ 
```

**Pull request merged**